wandb/util.py

```diff
@@ -1467,9 +1467,63 @@ def check_dict_contains_nested_artifact(d: dict, nested: bool = False) -> bool:
     return False
 
 
+def load_as_json_file_or_load_dict_as_json(config: str) -> Any:
+    if os.path.splitext(config)[-1] == ".json":
+        with open(config, "r") as f:
+            return json.load(f)
+    else:
+        try:
+            return json.loads(config)
+        except ValueError:
+            return None
+
+
 def _is_artifact_string_or_artifact(v):
     return (
         (isinstance(v, six.string_types) and v.startswith("wandb-artifact://"))
         or isinstance(v, wandb.Artifact)
         or isinstance(v, wandb.apis.public.Artifact)
     )
+
+
+def parse_artifact_string(v: str) -> Tuple[str, str]:
+    if not v.startswith("wandb-artifact://") and not v.startswith(
+        "wandb-artifact-digest://"
+    ):
+        raise ValueError(f"Invalid artifact string: {v}")
+
+    if v.startswith("wandb-artifact-digest://"):
+        return parse_artifact_digest_string(v)
+
+    parsed_v = v[len("wandb-artifact://") :]
+    # base_uri = None
+    # if is_uri(v[0]):
+    #     base_uri = v[0]
+    #     v = v[1:]
+
+    parts = parsed_v.split("/")
+    if len(parts) < 3:
+        raise ValueError(f"Invalid artifact string: {v}")
+    if len(parts) > 3:
+        # for now can't fetch paths but this will be supported in the future
+        # when we allow passing typed media objects
+        entity, project, name_and_alias_or_version = parts[:2]
+        return f"{entity}/{project}/{name_and_alias_or_version}"
+
+
+def parse_artifact_digest_string(v: str) -> str:
+    if not v.startswith("wandb-artifact-wandb-artifact-digest://"):
```

Comment:
Double `wandb-artifact`?

Reply:
Ah, this whole function should be removed